labview-interop/src/memory.rs

```diff
@@ -219,34 +252,58 @@ mod lv_owned {
         /// * This will create a handle to un-initialized memory. The provided initialisation
         ///    routine must prepare the memory.
         pub(crate) unsafe fn new_unsized(
-            init_routine: impl FnOnce(&mut UHandle<T>) -> Result<()>,
+            init_routine: impl FnOnce(&mut UHandle<'a, T>) -> Result<()>,
         ) -> Result<Self> {
             let handle = memory_api()?.new_handle(0);
             if handle.is_null() {
                 Err(LVInteropError::HandleCreationFailed)
             } else {
-                let mut new_value = Self(UHandle(handle as *mut *mut T));
+                let mut new_value = Self(UHandle(handle as *mut *mut T, PhantomData));
                 init_routine(&mut new_value)?;
                 Ok(new_value)
             }
         }
+
+        /// TODO test
+        /// Return the UHandle to the owned memory
+        ///
+        /// # Safety
+        ///
+        /// * This needs to take a mutable reference to self and lifetime annotation on UHandle,
+        ///    in order to avoid creating multiple UHandles.
+        pub fn handle(&'a mut self) -> UHandle<'a, T> {
+            UHandle(self.0 .0, PhantomData)
+        }
     }
 
-    impl<T: ?Sized> Deref for LvOwned<T> {
-        type Target = UHandle<T>;
+    /// TODO
+    /// potentially expensive operation
+    #[cfg(feature = "link")]
+    impl<'a, T: ?Sized> Clone for LvOwned<'a, T> {
+        fn clone(&self) -> Self {
+            let mut cloned_handle = UHandle(std::ptr::null_mut() as *mut *mut T, PhantomData);
+            unsafe {
+                self.clone_into_pointer(&mut cloned_handle as *mut UHandle<T>);
```

Comment:
Happy to take as is for now and test if this works with a null handle in. 

However I may refactor this to use the existing new method and provide an init function to do the copy so we have a single implementation of setting up a new owned handle.